diff --git a/www/articles.php b/www/articles.php
index 77a5f46..618d71e 100644
--- a/www/articles.php
+++ b/www/articles.php
@@ -9,14 +9,36 @@
  */
 
 // update message version - always use a string to avoid localized floats!
 $updateVersion = "47.1";
 
 //  xdebug_start_profiling();
 
+// HACK LJ >>>>>>>>>
+  if (isset($_SERVER['PATH_INFO'])) {
+    if (preg_match("!^/(en|fr)/manu(a|e)l\-!", $_SERVER['PATH_INFO'])) {
+      header("Location: http://docs.jelix.org".$_SERVER['PATH_INFO'], true, 301);
+      exit;
+    }
+  }
+
+  $lang ='en';
+  if(isset($_SERVER['PATH_INFO']) && (!isset($_REQUEST['do']) ||$_REQUEST['do'] != 'search')){
+    $_REQUEST['id'] = str_replace('/',':',$_SERVER['PATH_INFO']);
+    if($_REQUEST['id']{0} == ':')
+      $_REQUEST['id'] = substr($_REQUEST['id'],1);
+  }
+  if(isset($_REQUEST['id'])){
+    if($_REQUEST['id'] == 'fr' || strpos($_REQUEST['id'],'fr:')===0){
+       $lang='fr';
+    }
+  }
+  define('DOKU_LANG', $lang);
+// <<<<<<<<<< HACK LJ
+
 if(!defined('DOKU_INC')) define('DOKU_INC', dirname(__FILE__).'/');
 
 // define all DokuWiki globals here (needed within test requests but also helps to keep track)
 global  $ACT,  $INPUT, $QUERY, $ID, $REV, $DATE_AT, $IDX,
         $DATE, $RANGE, $HIGH, $TEXT, $PRE, $SUF, $SUM, $INFO, $JSINFO;
 
 
@@ -34,14 +56,26 @@ if(isset($_SERVER['HTTP_X_DOKUWIKI_DO'])) {
 require_once(DOKU_INC.'inc/init.php');
 
 //import variables
 $INPUT->set('id', str_replace("\xC2\xAD", '', $INPUT->str('id'))); //soft-hyphen
 $QUERY          = trim($INPUT->str('id'));
 $ID             = getID();
 
+// HACK LJ >>>>>>>>>
+if($ACT == 'show' && !isset($_GET['noredirect']) && $_SERVER["REQUEST_METHOD"] =='GET') {
+    $file = metaFN($ID, '.redirect');
+    if(@file_exists($file) && ($redirect = file_get_contents($file)) != '') {
+        $url = '/articles/'.str_replace(':','/', $redirect);
+        header("HTTP/1.0 301 Redirect");
+        header('Location: '.$url);
+        exit();
+    }
+}
+// <<<<<<<<<< HACK LJ
+
 $REV   = $INPUT->int('rev');
 $DATE_AT = $INPUT->str('at');
 $IDX   = $INPUT->str('idx');
 $DATE  = $INPUT->int('date');
 $RANGE = $INPUT->str('range');
 $HIGH  = $INPUT->param('s');
 if(empty($HIGH)) $HIGH = getGoogleQuery();
diff --git a/www/inc/form.php b/www/inc/form.php
index 91a1715..266e4c7 100644
--- a/www/inc/form.php
+++ b/www/inc/form.php
@@ -374,15 +374,15 @@ function form_makeCloseTag($tag) {
  * @param   array   $attrs  Optional attributes.
  * @return  array   pseudo-tag
  *
  * @author  Tom N Harris <tnharris@whoopdedo.org>
  */
 function form_makeWikiText($text, $attrs=array()) {
     $elem = array('_elem'=>'wikitext', '_text'=>$text,
-                        'class'=>'edit', 'cols'=>'80', 'rows'=>'10');
+                        'class'=>'edit', 'cols'=>'110', 'rows'=>'15');
     return array_merge($elem, $attrs);
 }
 
 /**
  * form_makeButton
  *
  * Create a form element for an action button.
diff --git a/www/inc/infoutils.php b/www/inc/infoutils.php
index fe312d1..dc0098c 100644
--- a/www/inc/infoutils.php
+++ b/www/inc/infoutils.php
@@ -54,17 +54,17 @@ function checkUpdateMessages(){
  * Return DokuWiki's version (split up in date and type)
  *
  * @author Andreas Gohr <andi@splitbrain.org>
  */
 function getVersionData(){
     $version = array();
     //import version string
-    if(file_exists(DOKU_INC.'VERSION')){
+    if(file_exists(DOKU_INC.'/../../lib/dokuwiki/VERSION')){
         //official release
-        $version['date'] = trim(io_readfile(DOKU_INC.'VERSION'));
+        $version['date'] = trim(io_readfile(DOKU_INC.'/../../lib/dokuwiki/VERSION'));
         $version['type'] = 'Release';
     }elseif(is_dir(DOKU_INC.'.git')){
         $version['type'] = 'Git';
         $version['date'] = 'unknown';
 
         $inventory = DOKU_INC.'.git/logs/HEAD';
         if(is_file($inventory)){
diff --git a/www/inc/lang/en/lang.php b/www/inc/lang/en/lang.php
index 9812ab6..13c461a 100644
--- a/www/inc/lang/en/lang.php
+++ b/www/inc/lang/en/lang.php
@@ -364,8 +364,29 @@ $lang['media_acl_warning']     = 'This list might not be complete due to ACL res
 
 $lang['currentns']             = 'Current namespace';
 $lang['searchresult']          = 'Search Result';
 $lang['plainhtml']             = 'Plain HTML';
 $lang['wikimarkup']            = 'Wiki Markup';
 $lang['page_nonexist_rev']     = 'Page did not exist at %s. It was subsequently created at <a href="%s">%s</a>.';
 $lang['unable_to_parse_date']  = 'Unable to parse at parameter "%s".';
+
+$lang['book_next_c'] = 'Next chapter';
+$lang['book_prev_c'] = 'Previous chapter';
+$lang['book_up_c'] = 'Chapter';
+$lang['book_current_c'] = 'Chapter';
+
+$lang['book_next_p'] = 'Next part';
+$lang['book_prev_p'] = 'Previous part';
+$lang['book_up_p'] = 'Part';
+$lang['book_current_p'] = 'Part';
+
+$lang['book_next_s'] = 'Next section';
+$lang['book_prev_s'] = 'Previous section';
+$lang['book_up_s'] = 'Section';
+$lang['book_current_s'] = 'Section';
+
+$lang['book_next_f'] = 'Next section';
+$lang['book_prev_f'] = 'Previous section';
+$lang['book_up_f'] = 'Section';
+$lang['book_current_f'] = 'Section';
+
 //Setup VIM: ex: et ts=2 :
diff --git a/www/inc/lang/fr/lang.php b/www/inc/lang/fr/lang.php
index c737862..ba11e19 100644
--- a/www/inc/lang/fr/lang.php
+++ b/www/inc/lang/fr/lang.php
@@ -365,7 +365,27 @@ $lang['media_restore']         = 'Restaurer cette version';
 $lang['media_acl_warning']     = 'En raison des restrictions dans les ACL et de pages cachées, cette liste peut ne pas être complète.';
 $lang['currentns']             = 'Catégorie courante';
 $lang['searchresult']          = 'Résultat de la recherche';
 $lang['plainhtml']             = 'HTML brut';
 $lang['wikimarkup']            = 'Wiki balise';
 $lang['page_nonexist_rev']     = 'La page n\'existait pas le %s. Elle a été créée le <a href="%s">%s</a>.';
 $lang['unable_to_parse_date']  = 'Ne peut analyser le paramètre date "%s".';
+
+$lang['book_next_c'] = 'Chapitre suivant';
+$lang['book_prev_c'] = 'Chapitre précédent ';
+$lang['book_up_c'] = 'Chapitre ';
+$lang['book_current_c'] = 'Chapitre ';
+
+$lang['book_next_p'] = 'Partie suivante';
+$lang['book_prev_p'] = 'Partie précédente ';
+$lang['book_up_p'] = 'Partie ';
+$lang['book_current_p'] = 'Partie ';
+
+$lang['book_next_s'] = 'Section suivante';
+$lang['book_prev_s'] = 'Section précédente ';
+$lang['book_up_s'] = 'Section parente';
+$lang['book_current_s'] = 'Section ';
+
+$lang['book_next_f'] = 'Section suivante';
+$lang['book_prev_f'] = 'Section précédente ';
+$lang['book_up_f'] = 'Section parente';
+$lang['book_current_f'] = 'Section ';
diff --git a/www/inc/pageutils.php b/www/inc/pageutils.php
index a5bf039..ff74e7f 100644
--- a/www/inc/pageutils.php
+++ b/www/inc/pageutils.php
@@ -32,15 +32,17 @@ function getID($param='id',$clean=true){
 
     //construct page id from request URI
     if(empty($id) && $conf['userewrite'] == 2){
         $request = $INPUT->server->str('REQUEST_URI');
         $script = '';
 
         //get the script URL
-        if($conf['basedir']){
+        if($_SERVER['PATH_INFO']) { // HACK_LJ
+            $id = substr($_SERVER['PATH_INFO'],1);
+        }elseif($conf['basedir']){ // end HACK_LJ
             $relpath = '';
             if($param != 'id') {
                 $relpath = 'lib/exe/';
             }
             $script = $conf['basedir'].$relpath.utf8_basename($INPUT->server->str('SCRIPT_FILENAME'));
 
         }elseif($INPUT->server->str('PATH_INFO')){
@@ -49,22 +51,24 @@ function getID($param='id',$clean=true){
             $script = $INPUT->server->str('SCRIPT_NAME');
         }elseif($INPUT->server->str('DOCUMENT_ROOT') && $INPUT->server->str('SCRIPT_FILENAME')){
             $script = preg_replace ('/^'.preg_quote($INPUT->server->str('DOCUMENT_ROOT'),'/').'/','',
                     $INPUT->server->str('SCRIPT_FILENAME'));
             $script = '/'.$script;
         }
 
-        //clean script and request (fixes a windows problem)
-        $script  = preg_replace('/\/\/+/','/',$script);
-        $request = preg_replace('/\/\/+/','/',$request);
+        if ($script) {// HACK_LJ
+            //clean script and request (fixes a windows problem)
+            $script  = preg_replace('/\/\/+/','/',$script);
+            $request = preg_replace('/\/\/+/','/',$request);
+            //remove script URL and Querystring to gain the id
+            if(preg_match('/^'.preg_quote($script,'/').'(.*)/',$request, $match)){
+                $id = preg_replace ('/\?.*/','',$match[1]);
+            }
+        }// end HACK_LJ
 
-        //remove script URL and Querystring to gain the id
-        if(preg_match('/^'.preg_quote($script,'/').'(.*)/',$request, $match)){
-            $id = preg_replace ('/\?.*/','',$match[1]);
-        }
         $id = urldecode($id);
         //strip leading slashes
         $id = preg_replace('!^/+!','',$id);
     }
 
     // Namespace autolinking from URL
     if(substr($id,-1) == ':' || ($conf['useslash'] && substr($id,-1) == '/')){
diff --git a/www/inc/parser/xhtml.php b/www/inc/parser/xhtml.php
index 9d7613f..720f1bb 100644
--- a/www/inc/parser/xhtml.php
+++ b/www/inc/parser/xhtml.php
@@ -215,17 +215,17 @@ class Doku_Renderer_xhtml extends Doku_Renderer {
         }
 
         // write the header
         $this->doc .= DOKU_LF.'<h'.$level;
         if($level <= $conf['maxseclevel']) {
             $this->doc .= ' class="'.$this->startSectionEdit($pos, 'section', $text).'"';
         }
-        $this->doc .= ' id="'.$hid.'">';
+        $this->doc .= '><a name="'.$hid.'" id="'.$hid.'">';
         $this->doc .= $this->_xmlEntities($text);
-        $this->doc .= "</h$level>".DOKU_LF;
+        $this->doc .= '<a title="Link to this section" class="anchor" href="#'.$hid.'"> ¶</a></h'.$level.'>'.DOKU_LF;
     }
 
     /**
      * Open a new section
      *
      * @param int $level section level (as determined by the previous header)
      */
diff --git a/www/inc/utf8.php b/www/inc/utf8.php
index 2b6a0c4..6f20e67 100644
--- a/www/inc/utf8.php
+++ b/www/inc/utf8.php
@@ -627,15 +627,15 @@ if(!function_exists('utf8_decode_numeric')){
                 $cp = intval($ent[3]);
                 break;
         }
         return unicode_to_utf8(array($cp));
     }
 }
 
-if(!class_exists('utf8_entity_decoder')){
+if(!class_exists('utf8_entity_decoder', false)){
     /**
      * Encapsulate HTML entity decoding tables
      */
     class utf8_entity_decoder {
         var $table;
 
         /**
