diff --git a/www/articles.php b/www/articles.php
index afaf79c..1c95590 100644
--- a/www/articles.php
+++ b/www/articles.php
@@ -13,6 +13,28 @@ $updateVersion = 44;
 
 //  xdebug_start_profiling();
 
+// HACK LJ >>>>>>>>>
+  if (isset($_SERVER['PATH_INFO'])) {
+    if (preg_match("!^/(en|fr)/manu(a|e)l\-!", $_SERVER['PATH_INFO'])) {
+      header("Location: http://docs.jelix.org".$_SERVER['PATH_INFO'], true, 301);
+      exit;
+    }
+  }
+
+  $lang ='en';
+  if(isset($_SERVER['PATH_INFO']) && (!isset($_REQUEST['do']) ||$_REQUEST['do'] != 'search')){
+    $_REQUEST['id'] = str_replace('/',':',$_SERVER['PATH_INFO']);
+    if($_REQUEST['id']{0} == ':')
+      $_REQUEST['id'] = substr($_REQUEST['id'],1);
+  }
+  if(isset($_REQUEST['id'])){
+    if($_REQUEST['id'] == 'fr' || strpos($_REQUEST['id'],'fr:')===0){
+       $lang='fr';
+    }
+  }
+  define('DOKU_LANG', $lang);
+// <<<<<<<<<< HACK LJ
+
 if(!defined('DOKU_INC')) define('DOKU_INC', dirname(__FILE__).'/');
 
 if(isset($_SERVER['HTTP_X_DOKUWIKI_DO'])) {
@@ -33,6 +55,16 @@ $INPUT->set('id', str_replace("\xC2\xAD", '', $INPUT->str('id'))); //soft-hyphen
 $QUERY          = trim($INPUT->str('id'));
 $ID             = getID();
 
+if($ACT == 'show' && !isset($_GET['noredirect']) && $_SERVER["REQUEST_METHOD"] =='GET') {
+    $file = metaFN($ID, '.redirect');
+    if(@file_exists($file) && ($redirect = file_get_contents($file)) != '') {
+        $url = '/articles/'.str_replace(':','/', $redirect);
+        header("HTTP/1.0 301 Redirect");
+        header('Location: '.$url);
+        exit();
+    }
+}
+
 $REV   = $INPUT->int('rev');
 $IDX   = $INPUT->str('idx');
 $DATE  = $INPUT->int('date');
diff --git a/www/inc/form.php b/www/inc/form.php
index 9cd0491..19c1cd9 100644
--- a/www/inc/form.php
+++ b/www/inc/form.php
@@ -362,7 +362,7 @@ function form_makeCloseTag($tag) {
  */
 function form_makeWikiText($text, $attrs=array()) {
     $elem = array('_elem'=>'wikitext', '_text'=>$text,
-                        'class'=>'edit', 'cols'=>'80', 'rows'=>'10');
+                        'class'=>'edit', 'cols'=>'110', 'rows'=>'15');
     return array_merge($elem, $attrs);
 }
 
diff --git a/www/inc/infoutils.php b/www/inc/infoutils.php
index 0992040..f0d2268 100644
--- a/www/inc/infoutils.php
+++ b/www/inc/infoutils.php
@@ -52,9 +52,9 @@ function checkUpdateMessages(){
 function getVersionData(){
     $version = array();
     //import version string
-    if(@file_exists(DOKU_INC.'VERSION')){
+    if(@file_exists(DOKU_INC.'/../../lib/dokuwiki/VERSION')){
         //official release
-        $version['date'] = trim(io_readfile(DOKU_INC.'VERSION'));
+        $version['date'] = trim(io_readfile(DOKU_INC.'/../../lib/dokuwiki/VERSION'));
         $version['type'] = 'Release';
     }elseif(is_dir(DOKU_INC.'.git')){
         $version['type'] = 'Git';
diff --git a/www/inc/lang/en/lang.php b/www/inc/lang/en/lang.php
index 5922891..86cf33d 100644
--- a/www/inc/lang/en/lang.php
+++ b/www/inc/lang/en/lang.php
@@ -366,4 +366,25 @@ $lang['currentns']             = 'Current namespace';
 $lang['searchresult']          = 'Search Result';
 $lang['plainhtml']             = 'Plain HTML';
 $lang['wikimarkup']            = 'Wiki Markup';
+
+$lang['book_next_c'] = 'Next chapter';
+$lang['book_prev_c'] = 'Previous chapter';
+$lang['book_up_c'] = 'Chapter';
+$lang['book_current_c'] = 'Chapter';
+
+$lang['book_next_p'] = 'Next part';
+$lang['book_prev_p'] = 'Previous part';
+$lang['book_up_p'] = 'Part';
+$lang['book_current_p'] = 'Part';
+
+$lang['book_next_s'] = 'Next section';
+$lang['book_prev_s'] = 'Previous section';
+$lang['book_up_s'] = 'Section';
+$lang['book_current_s'] = 'Section';
+
+$lang['book_next_f'] = 'Next section';
+$lang['book_prev_f'] = 'Previous section';
+$lang['book_up_f'] = 'Section';
+$lang['book_current_f'] = 'Section';
+
 //Setup VIM: ex: et ts=2 :
diff --git a/www/inc/lang/fr/lang.php b/www/inc/lang/fr/lang.php
index 1f11608..4e544cc 100644
--- a/www/inc/lang/fr/lang.php
+++ b/www/inc/lang/fr/lang.php
@@ -358,3 +358,23 @@ $lang['currentns']             = 'Namespace actuel';
 $lang['searchresult']          = 'Résultat de la recherche';
 $lang['plainhtml']             = 'HTML brut';
 $lang['wikimarkup']            = 'Wiki balise';
+
+$lang['book_next_c'] = 'Chapitre suivant';
+$lang['book_prev_c'] = 'Chapitre précédent ';
+$lang['book_up_c'] = 'Chapitre ';
+$lang['book_current_c'] = 'Chapitre ';
+
+$lang['book_next_p'] = 'Partie suivante';
+$lang['book_prev_p'] = 'Partie précédente ';
+$lang['book_up_p'] = 'Partie ';
+$lang['book_current_p'] = 'Partie ';
+
+$lang['book_next_s'] = 'Section suivante';
+$lang['book_prev_s'] = 'Section précédente ';
+$lang['book_up_s'] = 'Section parente';
+$lang['book_current_s'] = 'Section ';
+
+$lang['book_next_f'] = 'Section suivante';
+$lang['book_prev_f'] = 'Section précédente ';
+$lang['book_up_f'] = 'Section parente';
+$lang['book_current_f'] = 'Section ';
diff --git a/www/inc/pageutils.php b/www/inc/pageutils.php
index 8474c56..0ee1689 100644
--- a/www/inc/pageutils.php
+++ b/www/inc/pageutils.php
@@ -32,7 +32,9 @@ function getID($param='id',$clean=true){
         $script = '';
 
         //get the script URL
-        if($conf['basedir']){
+        if($_SERVER['PATH_INFO']) { // HACK_LJ
+            $id = substr($_SERVER['PATH_INFO'],1);
+        }elseif($conf['basedir']){ // end HACK_LJ
             $relpath = '';
             if($param != 'id') {
                 $relpath = 'lib/exe/';
@@ -49,14 +51,16 @@ function getID($param='id',$clean=true){
             $script = '/'.$script;
         }
 
-        //clean script and request (fixes a windows problem)
-        $script  = preg_replace('/\/\/+/','/',$script);
-        $request = preg_replace('/\/\/+/','/',$request);
+        if ($script) {// HACK_LJ
+            //clean script and request (fixes a windows problem)
+            $script  = preg_replace('/\/\/+/','/',$script);
+            $request = preg_replace('/\/\/+/','/',$request);
+            //remove script URL and Querystring to gain the id
+            if(preg_match('/^'.preg_quote($script,'/').'(.*)/',$request, $match)){
+                $id = preg_replace ('/\?.*/','',$match[1]);
+            }
+        }// end HACK_LJ
 
-        //remove script URL and Querystring to gain the id
-        if(preg_match('/^'.preg_quote($script,'/').'(.*)/',$request, $match)){
-            $id = preg_replace ('/\?.*/','',$match[1]);
-        }
         $id = urldecode($id);
         //strip leading slashes
         $id = preg_replace('!^/+!','',$id);
diff --git a/www/inc/parser/xhtml.php b/www/inc/parser/xhtml.php
index cf36a81..8530fa9 100644
--- a/www/inc/parser/xhtml.php
+++ b/www/inc/parser/xhtml.php
@@ -176,9 +176,9 @@ class Doku_Renderer_xhtml extends Doku_Renderer {
         if ($level <= $conf['maxseclevel']) {
             $this->doc .= ' class="' . $this->startSectionEdit($pos, 'section', $text) . '"';
         }
-        $this->doc .= ' id="'.$hid.'">';
+        $this->doc .= '><a name="'.$hid.'" id="'.$hid.'">';
         $this->doc .= $this->_xmlEntities($text);
-        $this->doc .= "</h$level>".DOKU_LF;
+        $this->doc .= '<a title="Link to this section" class="anchor" href="#'.$hid.'"> ¶</a></h'.$level.'>'.DOKU_LF;
     }
 
     function section_open($level) {
diff --git a/www/inc/utf8.php b/www/inc/utf8.php
index c944667..fe840b8 100644
--- a/www/inc/utf8.php
+++ b/www/inc/utf8.php
@@ -580,7 +580,7 @@ if(!function_exists('utf8_decode_numeric')){
     }
 }
 
-if(!class_exists('utf8_entity_decoder')){
+if(!class_exists('utf8_entity_decoder', false)){
     /**
      * Encapsulate HTML entity decoding tables
      */
